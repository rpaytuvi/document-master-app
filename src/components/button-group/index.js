import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View} from 'react-native';
import {GhostButton} from '../';
import Styles from './styles';

class ButtonGroup extends Component {
  static propTypes = {
    textLeft: PropTypes.string,
    iconLeft: PropTypes.string,
    textRight: PropTypes.string,
    iconRight: PropTypes.string,
    onPressLeft: PropTypes.func.isRequired,
    onPressRight: PropTypes.func.isRequired,
    leftBigger: PropTypes.bool,
    selectedIndex: PropTypes.number,
  };

  static defaultProps = {
    textLeft: '',
    iconLeft: '',
    textRight: PropTypes.string,
    iconRight: PropTypes.string,
    onPressLeft: () => {},
    onPressRight: () => {},
    leftBigger: false,
    selectedIndex: -1,
  };

  render = () => {
    return (
      <View style={Styles.container}>
        <View
          style={
            this.props.leftBigger
              ? Styles.buttonContainerBig
              : Styles.buttonContainer
          }>
          <GhostButton
            text={this.props.textLeft}
            icon={this.props.iconLeft}
            onPress={this.props.onPressLeft}
            customStyle={Object.assign(
              {},
              Styles.leftButton,
              this.props.selectedIndex > -1 ? Styles.buttonUnselected : {},
              this.props.selectedIndex === 0 ? Styles.buttonSelected : {},
            )}
            customTextStyle={Styles.buttonText}
            customIconStyle={Object.assign(
              {},
              Styles.buttonIcon,
              this.props.selectedIndex === 0 ? Styles.buttonIconSelected : {},
            )}
          />
        </View>
        <View style={Styles.buttonContainer}>
          <GhostButton
            text={this.props.textRight}
            icon={this.props.iconRight}
            onPress={this.props.onPressRight}
            customStyle={Object.assign(
              {},
              Styles.rightButton,
              this.props.selectedIndex > -1 ? Styles.buttonUnselected : {},
              this.props.selectedIndex === 1 ? Styles.buttonSelected : {},
            )}
            customTextStyle={Styles.buttonText}
            customIconStyle={Object.assign(
              {},
              Styles.buttonIcon,
              this.props.selectedIndex === 1 ? Styles.buttonIconSelected : {},
            )}
          />
        </View>
      </View>
    );
  };
}

export default ButtonGroup;
