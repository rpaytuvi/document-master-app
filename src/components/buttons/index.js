import GhostButton from './ghost';
import HeaderButton from './header';
import MainButton from './main';

export {GhostButton, HeaderButton, MainButton};
