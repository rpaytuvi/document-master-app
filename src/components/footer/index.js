import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View} from 'react-native';
import {MainButton} from '../';
import Styles from './styles';

class Footer extends Component {
  static propTypes = {
    buttonText: PropTypes.string.isRequired,
    buttonIcon: PropTypes.string,
    action: PropTypes.func.isRequired,
  };

  static defaultProps = {
    buttonText: '',
    buttonIcon: '',
    action: () => {},
  };

  render = () => {
    return (
      <View style={Styles.container}>
        <MainButton
          text={this.props.buttonText}
          icon={this.props.buttonIcon}
          onPress={this.props.action}
        />
      </View>
    );
  };
}

export default Footer;
