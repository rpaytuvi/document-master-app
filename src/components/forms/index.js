import DocumentForm from './document';

export {DocumentForm};
