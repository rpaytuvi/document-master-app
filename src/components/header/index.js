import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {Text, View} from 'react-native';
import {HeaderButton} from '../';
import Styles from './styles';

class Header extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    action: PropTypes.func.isRequired,
    withAlert: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    title: '',
    action: () => {},
    withAlert: false,
  };

  render = () => {
    return (
      <View style={Styles.container}>
        <View style={Styles.titleContainer}>
          <Text style={Styles.title}>{this.props.title}</Text>
        </View>
        <View style={Styles.buttonContainer}>
          <HeaderButton
            onPress={this.props.action}
            withAlert={this.props.withAlert}
          />
        </View>
      </View>
    );
  };
}

export default Header;
