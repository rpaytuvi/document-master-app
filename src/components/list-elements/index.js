import DocumentLargeListElement from './document-large';
import DocumentSmallListElement from './document-small';

export {DocumentLargeListElement, DocumentSmallListElement};
