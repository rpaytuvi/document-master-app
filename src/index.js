import React, {Component} from 'react';
import {Provider} from 'react-redux';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {Root} from './screens';
import store from './redux';

class App extends Component {
  render = () => {
    return (
      <SafeAreaProvider>
        <Provider store={store}>
          <Root />
        </Provider>
      </SafeAreaProvider>
    );
  };
}

export default App;
