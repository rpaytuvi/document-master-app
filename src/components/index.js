import {HeaderButton, MainButton} from './buttons';
import {
  DocumentLargeListElement,
  DocumentSmallListElement,
} from './list-elements';
import {DocumentsListHeader} from './list-headers';
import {DocumentsList} from './lists';
import Footer from './footer';
import Header from './header';

export {
  HeaderButton,
  MainButton,
  DocumentLargeListElement,
  DocumentSmallListElement,
  DocumentsListHeader,
  DocumentsList,
  Footer,
  Header,
};
