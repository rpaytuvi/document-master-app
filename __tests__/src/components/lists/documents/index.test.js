import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {DocumentsList} from '../../../../../src/components';

test('renders correctly', () => {
  const tree = renderer.create(<DocumentsList />).toJSON();
  expect(tree).toMatchSnapshot();
});
