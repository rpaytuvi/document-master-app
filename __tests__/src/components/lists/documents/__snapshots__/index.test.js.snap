// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 4,
            "elevation": 5,
            "flexDirection": "column",
            "margin": 32,
            "padding": 16,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderColor": "#e8e9ec",
                "borderRadius": 4,
                "borderWidth": 0,
                "flex": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <Icon
              name="close"
              size={24}
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "#646d7a",
                    "marginRight": 8,
                  },
                  Object {
                    "marginRight": 0,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#646d7a",
                    "fontSize": 20,
                    "fontWeight": "600",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "marginBottom": 8,
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#3f82f2",
                  "borderRadius": 12,
                  "borderWidth": 2,
                  "height": 24,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#3f82f2",
                    "borderRadius": 6,
                    "height": 12,
                    "width": 12,
                  }
                }
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#646d7a",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                  Object {},
                ]
              }
            >
              Title
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "marginBottom": 8,
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#3f82f2",
                  "borderRadius": 12,
                  "borderWidth": 2,
                  "height": 24,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "width": 24,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#646d7a",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                  Object {},
                ]
              }
            >
              Version
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "marginBottom": 8,
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#3f82f2",
                  "borderRadius": 12,
                  "borderWidth": 2,
                  "height": 24,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "width": 24,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#646d7a",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                  Object {},
                ]
              }
            >
              Created
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "marginBottom": 8,
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#3f82f2",
                  "borderRadius": 12,
                  "borderWidth": 2,
                  "height": 24,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "width": 24,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#646d7a",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                  Object {},
                ]
              }
            >
              Updated
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>,
  <RCTScrollView
    ListEmptyComponent={
      <View
        style={
          Object {
            "marginTop": 80,
            "padding": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#6f7583",
              "fontSize": 24,
              "textAlign": "center",
            }
          }
        >
          No documents to display
        </Text>
      </View>
    }
    ListHeaderComponent={
      <DocumentsListHeader
        changeView={[Function]}
        openSortBy={[Function]}
        selectedView="list"
        sortDirection="desc"
        toggleSortDirection={[Function]}
      />
    }
    contentContainerStyle={
      Object {
        "paddingVertical": 16,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
              "marginBottom": 16,
              "paddingHorizontal": 16,
            }
          }
        >
          <View
            style={
              Object {
                "width": 140,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 2,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderBottomRightRadius": 0,
                      "borderColor": "#e8e9ec",
                      "borderRadius": 4,
                      "borderTopRightRadius": 0,
                      "borderWidth": 2,
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "opacity": 1,
                      "padding": 8,
                    }
                  }
                >
                  <Icon
                    name="unfold-more"
                    size={24}
                    style={
                      Array [
                        Object {
                          "alignSelf": "center",
                          "color": "#646d7a",
                          "marginRight": 8,
                        },
                        Object {
                          "marginRight": 0,
                        },
                      ]
                    }
                  />
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#646d7a",
                          "fontSize": 20,
                          "fontWeight": "600",
                        },
                        Object {
                          "fontSize": 16,
                        },
                      ]
                    }
                  >
                    Sort by
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderBottomLeftRadius": 0,
                      "borderColor": "#e8e9ec",
                      "borderLeftWidth": 0,
                      "borderRadius": 4,
                      "borderTopLeftRadius": 0,
                      "borderWidth": 2,
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "opacity": 1,
                      "padding": 8,
                    }
                  }
                >
                  <Icon
                    name="keyboard-arrow-down"
                    size={24}
                    style={
                      Array [
                        Object {
                          "alignSelf": "center",
                          "color": "#646d7a",
                          "marginRight": 8,
                        },
                        Object {
                          "marginRight": 0,
                        },
                      ]
                    }
                  />
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#646d7a",
                          "fontSize": 20,
                          "fontWeight": "600",
                        },
                        Object {
                          "fontSize": 16,
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "width": 140,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderBottomRightRadius": 0,
                      "borderColor": "#e8e9ec",
                      "borderRadius": 4,
                      "borderTopRightRadius": 0,
                      "borderWidth": 2,
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "opacity": 1,
                      "padding": 8,
                    }
                  }
                >
                  <Icon
                    name="list"
                    size={24}
                    style={
                      Array [
                        Object {
                          "alignSelf": "center",
                          "color": "#646d7a",
                          "marginRight": 8,
                        },
                        Object {
                          "color": "#3f82f2",
                          "marginRight": 0,
                        },
                      ]
                    }
                  />
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#646d7a",
                          "fontSize": 20,
                          "fontWeight": "600",
                        },
                        Object {
                          "fontSize": 16,
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#f4f4f6",
                      "borderBottomLeftRadius": 0,
                      "borderColor": "#e8e9ec",
                      "borderLeftWidth": 0,
                      "borderRadius": 4,
                      "borderTopLeftRadius": 0,
                      "borderWidth": 2,
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "opacity": 1,
                      "padding": 8,
                    }
                  }
                >
                  <Icon
                    name="grid-view"
                    size={24}
                    style={
                      Array [
                        Object {
                          "alignSelf": "center",
                          "color": "#646d7a",
                          "marginRight": 8,
                        },
                        Object {
                          "marginRight": 0,
                        },
                      ]
                    }
                  />
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#646d7a",
                          "fontSize": 20,
                          "fontWeight": "600",
                        },
                        Object {
                          "fontSize": 16,
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "marginTop": 80,
            "padding": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#6f7583",
              "fontSize": 24,
              "textAlign": "center",
            }
          }
        >
          No documents to display
        </Text>
      </View>
    </View>
  </RCTScrollView>,
]
`;
